### PR TITLE
Feature/oasis

### PR DESCRIPTION
**Pull request by Entodi**

### CODE REVIEW

File: fusion/criterion/loss/dim/xx_dim.py [hunk 1]
```diff
@@ -16,16 +16,15 @@ def __call__(self, reps, convs):
...
                         loss = self._weight * loss
                         total_loss, raw_losses = self._update_loss(
                             name, total_loss, raw_losses, loss, penalty
                         )
         return total_loss, raw_losses
 
     def _name_it(self, rep_source_id, conv_source_id, dim_conv):
-        return f"{self._name}{dim_conv}_" \
-               f"{rep_source_id}_{conv_source_id}"
+        return f"{self._name}{dim_conv}_" f"{rep_source_id}_{conv_source_id}"
```
Comment (Cybsloth) [inline fusion/criterion/loss/dim/xx_dim.py:30]: to single format string

File: fusion/dataset/oasis/__init__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from .oasis import Oasis
+
+
+__all__ = [
+    "Oasis",
+]
```
Comment (Cybsloth) [inline fusion/dataset/oasis/__init__.py:6]: To single line

File: fusion/runner/catalyst/oasis_runner.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from .catalyst_runner import CatalystRunner
+
+
+class OasisRunner(CatalystRunner):
+    def _unpack_batch(self, batch):
+        x = [
+            v['data'] for k, v in batch.items() if not k.startswith('label')
+        ]
+        y = batch["label"]
+        return x, y
```
Comment (Cybsloth) [inline fusion/runner/catalyst/oasis_runner.py:10]: "No newline at end of file" - warning

File: fusion/runner/runner_provider.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from .catalyst import CatalystRunner, MnistSvhnRunner, OasisRunner
+from fusion.utils import ObjectProvider
+
+
+runner_provider = ObjectProvider()
+runner_provider.register_object("CatalystRunner", CatalystRunner)
+runner_provider.register_object("MnistSvhnRunner", MnistSvhnRunner)
+runner_provider.register_object("OasisRunner", OasisRunner)
```
Comment (Cybsloth) [inline fusion/runner/runner_provider.py:8]: "No newline at end of file" - warning
